python/tester/upload_hsc_rc2.py

```diff
@@ -57,26 +78,37 @@ def main():
     date = time.strftime("%Y%m%d")
 
     kafka_url = "https://usdf-rsp-dev.slac.stanford.edu/sasquatch-rest-proxy/topics/test.next-visit"
-    endpoint_url = "https://s3dfrgw.slac.stanford.edu"
-    s3 = boto3.resource("s3", endpoint_url=endpoint_url)
-    dest_bucket = s3.Bucket("rubin:rubin-pp")
-    dest_bucket.meta.client.meta.events.unregister("before-parameter-build.s3", validate_bucket_name)
+    _set_s3_bucket()
 
     last_group = get_last_group(dest_bucket, "HSC", date)
     group_num = last_group + random.randrange(10, 19)
     _log.debug(f"Last group {last_group}; new group base {group_num}")
 
     butler = Butler("/repo/main")
     visit_list = get_hsc_visit_list(butler, n_groups)
-    for visit in visit_list:
-        group_num += 1
-        _log.info(f"Slewing to group {group_num}, with HSC visit {visit}")
-        time.sleep(SLEW_INTERVAL)
-        refs = prepare_one_visit(kafka_url, str(group_num), butler, visit)
-        _log.info(f"Taking exposure for group {group_num}")
-        time.sleep(EXPOSURE_INTERVAL)
-        _log.info(f"Uploading detector images for group {group_num}")
-        upload_hsc_images(dest_bucket, str(group_num), butler, refs)
+
+    # fork pools don't work well with connection pools, such as those used
+    # for Butler registry or S3.
+    context = multiprocessing.get_context("spawn")
+    max_processes = _get_max_processes()
+    # Use a shared pool to minimize initialization overhead. This has the
+    # benefit of letting the pool be initialized in parallel with the first
+    # exposure.
+    _log.debug("Uploading with %d processes...", max_processes)
+    with context.Pool(processes=max_processes, initializer=_set_s3_bucket) as pool, \
+            tempfile.TemporaryDirectory() as temp_dir:
+        for visit in visit_list:
+            group_num += 1
+            _log.info(f"Slewing to group {group_num}, with HSC visit {visit}")
+            time.sleep(SLEW_INTERVAL)
+            refs = prepare_one_visit(kafka_url, str(group_num), butler, visit)
+            _log.info(f"Taking exposure for group {group_num}")
+            time.sleep(EXPOSURE_INTERVAL)
+            _log.info(f"Uploading detector images for group {group_num}")
+            upload_hsc_images(pool, temp_dir, str(group_num), butler, refs)
```

Comment:
Is this the correct order? Slew, _then_ alert, then immediately expose? `upload.py` does the same thing.